crates/extensions/c8y_http_proxy/src/credentials.rs

```diff
@@ -32,16 +32,17 @@ impl C8YJwtRetriever {
 
 #[async_trait]
 impl Server for C8YJwtRetriever {
```

Comment:
Adding the new `C8YBasicAuthProvider` in this PR would really help to validate the refactoring, even if not use yet.

Reply:
Sorry, I should have remove this comment which is conflicting with my other comment https://github.com/thin-edge/thin-edge.io/pull/3190#pullrequestreview-2374370308, telling that:

> the difference between the misc auth method has to be addressed in the c8y_api with a facade over the `C8yMqttJwtTokenRetriever` and a `BasicAuthRetriver`. Both returning an authentication header ready to be used.

Anyway. This can be done in the next step.